### ISSUE TITLE
tl-wdr3600 build fails

### ISSUE THREAD

**Issue by bsdhsr** [freebsd/freebsd-wifi-build#113]
Also tried with 11-STABLE sources.

Command:

`~/freebsd-wifi/freebsd-wifi-build/build/bin/build tl-wdr3600`

--------------------------------------------------------------
>>> stage 4.2: building libraries
--------------------------------------------------------------
===> gnu/lib/libssp/libssp_nonshared (obj,all,install)
===> gnu/lib/libgcc (obj,all,install)
===> lib/libcompiler_rt (obj,all,install)
===> gnu/lib/csu (obj,all,install)
===> lib/csu (obj,all,install)
===> lib/csu/mips (obj)
===> lib/csu/mips (all)
===> lib/csu/mips (install)
===> lib/libcompiler_rt (obj,all,install)
{standard input}: Assembler messages:
{standard input}:87: Warning: no .frame pseudo-op used in PIC code
{standard input}: Assembler messages:
{standard input}:79: Warning: no .frame pseudo-op used in PIC code
{standard input}: Assembler messages:
{standard input}:102: Warning: no .frame pseudo-op used in PIC code
===> lib/libc (obj,all,install)
{standard input}: Assembler messages:
{standard input}:98: Warning: no .frame pseudo-op used in PIC code
{standard input}: Assembler messages:
{standard input}:79: Warning: no .frame pseudo-op used in PIC code
===> lib/libc_nonshared (obj,all,install)
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getgrent.c: In function 'grp_marshal_func':
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getgrent.c:215:9: warning: variable 'orig_buf_size' set but not used [-Wunused-but-set-variable]
  size_t orig_buf_size;
         ^~~~~~~~~~~~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getgrent.c:214:8: warning: variable 'orig_buf' set but not used [-Wunused-but-set-variable]
  char *orig_buf;
        ^~~~~~~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getgrent.c:212:8: warning: variable 'gid' set but not used [-Wunused-but-set-variable]
  gid_t gid;
        ^~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getgrent.c:211:8: warning: variable 'name' set but not used [-Wunused-but-set-variable]
  char *name;
        ^~~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getgrent.c: In function 'grp_unmarshal_func':
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getgrent.c:307:8: warning: variable 'gid' set but not used [-Wunused-but-set-variable]
  gid_t gid;
        ^~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getgrent.c:306:8: warning: variable 'name' set but not used [-Wunused-but-set-variable]
  char *name;
        ^~~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getpwent.c: In function 'pwd_marshal_func':
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getpwent.c:264:9: warning: variable 'orig_buf_size' set but not used [-Wunused-but-set-variable]
  size_t orig_buf_size;
         ^~~~~~~~~~~~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getpwent.c:263:8: warning: variable 'orig_buf' set but not used [-Wunused-but-set-variable]
  char *orig_buf;
        ^~~~~~~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getpwent.c:261:8: warning: variable 'uid' set but not used [-Wunused-but-set-variable]
  uid_t uid;
        ^~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getpwent.c:260:8: warning: variable 'name' set but not used [-Wunused-but-set-variable]
  char *name;
        ^~~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getpwent.c: In function 'pwd_unmarshal_func':
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getpwent.c:365:8: warning: variable 'uid' set but not used [-Wunused-but-set-variable]
  uid_t uid;
        ^~~
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/gen/getpwent.c:364:8: warning: variable 'name' set but not used [-Wunused-but-set-variable]
  char *name;
        ^~~~
{standard input}: Assembler messages:
{standard input}:10: Warning: no .frame pseudo-op used in PIC code
/usr/home/hsr/freebsd-wifi/src/11-release/lib/libc/isc/ev_streams.c:23:19: error: 'rcsid' defined but not used [-Werror=unused-const-variable=]
 static const char rcsid[] = "$Id: ev_streams.c,v 1.5 2005/04/27 04:56:36 sra Exp $";
                   ^~~~~
cc1: all warnings being treated as errors
--- ev_streams.o ---
*** [ev_streams.o] Error code 1

make[4]: stopped in /usr/home/hsr/freebsd-wifi/src/11-release/lib/libc
1 error

make[4]: stopped in /usr/home/hsr/freebsd-wifi/src/11-release/lib/libc
--- lib/libc__L ---
*** [lib/libc__L] Error code 2

make[3]: stopped in /usr/home/hsr/freebsd-wifi/src/11-release
1 error

make[3]: stopped in /usr/home/hsr/freebsd-wifi/src/11-release
--- libraries ---
*** [libraries] Error code 2

make[2]: stopped in /usr/home/hsr/freebsd-wifi/src/11-release
1 error

make[2]: stopped in /usr/home/hsr/freebsd-wifi/src/11-release
--- _libraries ---
*** [_libraries] Error code 2

make[1]: stopped in /usr/home/hsr/freebsd-wifi/src/11-release
1 error

make[1]: stopped in /usr/home/hsr/freebsd-wifi/src/11-release
--- buildworld ---
*** [buildworld] Error code 2

make: stopped in /usr/home/hsr/freebsd-wifi/src/11-release
1 error

make: stopped in /usr/home/hsr/freebsd-wifi/src/11-release

**Comment by Cka3o4Huk** [comment]
It's better to try 12-current  as well ;)

**Comment by croquagei** [comment]
Still unable to build tl-wdr3600.
I tried with 10.3, 11.0.1 and head (2017-07-11).

All builds failed at different times.  Included are links to build output:

[10.3 - Built world, failed to start building kernel, unable to find a kernel configuration file ](https://gist.github.com/croquagei/c55c1ea6498cc98c861d1c8d9df101d6)
[11.0.1 - Failed at building system libraries, about the time it started libc](https://gist.github.com/croquagei/16b927cfab0f6b553c68728f299297cc)
[Head - Built world, failed to build the kernel](https://gist.github.com/croquagei/53a386290c48a1144672d0a737456e28)

**Comment by erikarn** [comment]
hihi

I'm still sorting it out! give me a tiny bit more time pretty please


-adrian

**Comment by croquagei** [comment]
No problems, appreciate all the work 👍

**Comment by croquagei** [comment]
Both world and kernel compile but i don't get a final iamge.

Instead I've got a few kernels.
kernel.TL-WDR4300  
kernel.TL-WDR4300.debug  
kernel.TL-WDR4300.tramp.bin  


(Here's my build output for head)[https://gist.github.com/croquagei/0c0081ffcceaa69d56669cd521b14d2c]

**Comment by erikarn** [comment]
nothing after including modules?


-a

**Comment by croquagei** [comment]
Unsure how to do that.  
I just followed the instructions on the wiki page for the device.  Which was just check out the src tree then run the build script.

[Here's another build attempt, this time i remembered to redirect stderr.](https://gist.github.com/croquagei/f974aeb09222b5f3308f6a764146f63a)

**Comment by erikarn** [comment]
Aha, I know what else needs merging! Gimme a bit of time.

Thanks!



-a

**Comment by croquagei** [comment]
Any update on progress?

**Comment by Cka3o4Huk** [comment]
Hi,

As of today, ~/freebsd-wifi/freebsd-wifi-build/build/bin/build tl-wdr3600 works fine (host - HEAD, src - HEAD). @croquagei , do you still have a problem?

**Comment by mky** [comment]
Hi,

The problem still exists. Today I tried to build FreeBSD-11.1-RELEASE and FreeBSD-11-STABLE for TL-WDR3600. Both fails on buildworld. I tried also patches from pull request #124 but it doesn't help. I attaching build logs, maybe it can help with fixing this issue.

Build host: FreeBSD-11.1-RELEASE-p6 (amd64)

Packages:
bison-3.0.4,1
dialog4ports-0.1.6
fakeroot-1.22
git-2.15.1
gmake-4.2.1_1
libtool-2.4.6
lzma-16.04
mips-binutils-2.28,1
mips-gcc-6.3.0
mips-xtoolchain-gcc-0.2
subversion-1.9.7
uboot-mkimage-2010.12
wget-1.19.2
[freebsd-11.1-release-tl-wdr3600.txt](https://github.com/freebsd/freebsd-wifi-build/files/1775086/freebsd-11.1-release-tl-wdr3600.txt)
[freebsd-11.1-release-tl-wdr3600-pullreq-124.txt](https://github.com/freebsd/freebsd-wifi-build/files/1775087/freebsd-11.1-release-tl-wdr3600-pullreq-124.txt)
[freebsd-11-stable-tl-wdr3600.txt](https://github.com/freebsd/freebsd-wifi-build/files/1775088/freebsd-11-stable-tl-wdr3600.txt)
[freebsd-11-stable-tl-wdr3600-pullreq-124.txt](https://github.com/freebsd/freebsd-wifi-build/files/1775089/freebsd-11-stable-tl-wdr3600-pullreq-124.txt)
